### PR TITLE
Instructor: enroll students: add a 'paste' option to the context menu #8972

### LINKED ISSUE #8972: Instructor: enroll students: add a 'paste' option to the context menu
v6.7.0

current: in the spreadsheet interface of the enroll page, user can use Ctrl+V to paste data.

problem: a user reported looking for the `Paste` option for a long time in the context menu before realizing he can use Ctrl+V

suggestion: add a `Paste (Ctrl + V)` to the context menu of the spreadsheet UI

image

### PR DESCRIPTION
**Pull request by tanhengyeow**
Fixes #8972 

**Outline of Solution**
1. Added new modal box in helper file for paste option in `Handsontable` context menu.
2. Added new paste option in `Handsontable` context menu.

### CODE REVIEW

**Comment by tanhengyeow** [comment]
@wkurniawan07 @tran-tien-dat This is ready for review. Changes are minimal and does not affect back-end logic and front-end tests.

File: src/main/webapp/dev/js/main/instructorCourseEnrollPage.js [hunk 1]
```diff
@@ -37,6 +38,13 @@ const handsontable = new Handsontable(container, {
         'remove_row',
         'undo',
         'redo',
+        {
+            key: 'paste',
+            name: 'Paste',
+            callback: function() {
+                showModalPasteBox();
```
Comment (tran-tien-dat) [inline src/main/webapp/dev/js/main/instructorCourseEnrollPage.js]: No need to wrap this around an anonymous function

Comment (tanhengyeow) [reply src/main/webapp/dev/js/main/instructorCourseEnrollPage.js]: I think it needs to. If I do just `callback: showModalPasteBox()`, the page always show the modal box even if I didn't do Right Click > click `Paste`.

Comment (tran-tien-dat) [reply src/main/webapp/dev/js/main/instructorCourseEnrollPage.js]: You are supposed to pass a function, so it shouldn't have the brackets

Comment (tanhengyeow) [reply src/main/webapp/dev/js/main/instructorCourseEnrollPage.js]: Done 😄

File: src/main/webapp/dev/js/main/instructorCourseEnrollHelper.js [hunk 2]
```diff
@@ -47,8 +55,21 @@ function getUpdatedData(spreadsheetDataRows) {
     return spreadsheetDataRows.map(row => row.split('|'));
 }
 
+/**
+ * Function to show modal box when user clicks on the 'paste' option in the Handsontable context menu.
```
Comment (damithc) [inline src/main/webapp/dev/js/main/instructorCourseEnrollHelper.js]: ~~Function to show~~ Shows ...

**Comment by tran-tien-dat** [comment]
@tanhengyeow please fix the lint error

File: src/main/webapp/dev/js/main/instructorCourseEnrollHelper.js [hunk 3]
```diff
@@ -47,8 +55,21 @@ function getUpdatedData(spreadsheetDataRows) {
...
+ */
+function showPasteModalBox() {
+    const messageText = 'Pasting data through the context menu is not supported due to security complications. '
```
Comment (damithc) [inline src/main/webapp/dev/js/main/instructorCourseEnrollHelper.js]: `due to browser restrictions`? We don't want to alarm the user.

**Comment by tanhengyeow** [comment]
Ready to merge :)

**Pull request merged**